lib/ecmascript.ts

```diff
@@ -5936,115 +5934,56 @@ export function AddDurationToOrSubtractDurationFromZonedDateTime(
+export function RoundNumberToIncrement(quantity: number, increment: number, mode: Temporal.RoundingMode) {
+  const quotient = MathTrunc(quantity / increment);
+  const remainder = quantity % increment;
+  const sign = quantity < 0 ? 'negative' : 'positive';
+  const r1 = MathAbs(quotient);
+  const r2 = r1 + 1;
+  const cmp = ComparisonResult(MathAbs(remainder * 2) - increment);
+  const even = r1 % 2 === 0;
+  const unsignedRoundingMode = GetUnsignedRoundingMode(mode, sign);
+  const rounded =
+    MathAbs(quantity) === r1 * increment ? r1 : ApplyUnsignedRoundingMode(r1, r2, cmp, even, unsignedRoundingMode);
+  return increment * (sign === 'positive' ? rounded : -rounded);
 }
 
-function RoundJSNumberToIncrement(quantity: number, increment: number, mode: Temporal.RoundingMode) {
-  let quotient = MathTrunc(quantity / increment);
-  const remainder = quantity % increment;
-  if (remainder === 0) return quantity;
-  const sign = remainder < 0 ? -1 : 1;
-  const tiebreaker = MathAbs(remainder * 2);
-  const tie = tiebreaker === increment;
-  const expandIsNearer = tiebreaker > increment;
-  switch (mode) {
-    case 'ceil':
-      if (sign > 0) quotient += sign;
-      break;
-    case 'floor':
-      if (sign < 0) quotient += sign;
-      break;
-    case 'expand':
-      // always expand if there is a remainder
-      quotient += sign;
-      break;
-    case 'trunc':
-      // no change needed, because divmod is a truncation
-      break;
-    case 'halfCeil':
-      if (expandIsNearer || (tie && sign > 0)) quotient += sign;
-      break;
-    case 'halfFloor':
-      if (expandIsNearer || (tie && sign < 0)) quotient += sign;
-      break;
-    case 'halfExpand':
-      // "half up away from zero"
-      if (expandIsNearer || tie) quotient += sign;
-      break;
-    case 'halfTrunc':
-      if (expandIsNearer) quotient += sign;
-      break;
-    case 'halfEven': {
-      if (expandIsNearer || (tie && quotient % 2 === 1)) quotient += sign;
-      break;
-    }
+// ts-prune-ignore-next TODO: remove this after tests are converted to TS
+export function RoundNumberToIncrementAsIfPositive(
+  quantityParam: JSBI | bigint,
+  incrementParam: JSBI | bigint,
+  mode: Temporal.RoundingMode
+) {
+  const quantity = ensureJSBI(quantityParam);
+  const increment = ensureJSBI(incrementParam);
+  const quotient = JSBI.divide(quantity, increment);
+  const remainder = JSBI.remainder(quantity, increment);
+  const unsignedRoundingMode = GetUnsignedRoundingMode(mode, 'positive');
+  let r1: JSBI, r2: JSBI;
+  if (JSBI.lessThan(quantity, ZERO)) {
+    r1 = JSBI.subtract(quotient, ONE);
+    r2 = quotient;
+  } else {
+    r1 = quotient;
+    r2 = JSBI.add(quotient, ONE);
   }
-  return quotient * increment;
+  const cmp = compare(abs(JSBI.multiply(remainder, TWO)), increment) * (JSBI.lessThan(quantity, ZERO) ? -1 : 1);
+  const even = isEven(r1);
+  const rounded = JSBI.equal(JSBI.multiply(quotient, increment), quantity)
```

Comment:
Why is this test not `JSBI.equal(remainder, ZERO)` ?

Reply:
Nice find.